pkg/controller/replicas.go

```diff
@@ -21,7 +21,15 @@ func desiredStsReplicas(group string, sts *v1.StatefulSet, all []*v1.StatefulSet
 
 	leaderReplicas := *leader.Spec.Replicas
 	if leaderReplicas > followerReplicas {
-		// Scale up is always allowed immediately
+		// Handle scale-up scenarios
+		annotations := sts.GetAnnotations()
+		onlyWhenReady, ok := annotations[config.RolloutLeaderReadyAnnotationKey]
+		if ok && onlyWhenReady == config.RolloutLeaderReadyAnnotationValue {
+			// We only scale up once all of the leader pods are ready. Otherwise we do nothing.
+			if leaderReplicas != leader.Status.ReadyReplicas {
+				return followerReplicas, nil
```

Comment:
Might be good to emit a log message here (with the same fields as used below) to make it clear that we're holding back scaling up because the leader StatefulSet isn't ready